Crashes when leds length is 3

### What happened?

WLed crashes when setting some effects like Fireworks 1D

### To Reproduce Bug

1. Set the Length of leds to 3
2. Set the effect to Fireworks 1D

### Expected Behavior

To work fine.

### Install Method

Binary from WLED.me

### What version of WLED?

0.14.4, 0.15.0-b4

### Which microcontroller/board are you seeing the problem on?

ESP32

### Relevant log/trace output

```shell
abort() was called at PC 0x4008f49c on core 0

ELF file SHA256: 0000000000000000

Backtrace: 0x4008a7b8:0x3ffb7710 0x4008ab15:0x3ffb7730 0x4008f49c:0x3ffb7750 0x4008f9b4:0x3ffb7770 0x4008600d:0x3ffb7790 0x4008603e:0x3ffb77b0 0x4008645d:0x3ffb77d0 0x4000beaf:0x3ffb77f0 0x40084081:0x3ffb7810 0x4018a2b1:0x3ffb7830 0x401a1ce9:0x3ffb7850 0x4019cf91:0x3ffb7870 0x4014dc73:0x3ffb7890 0x4014dcdb:0x3ffb78b0 0x4008c55e:0x3ffb78d0

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DOUT, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:10124
load:0x40080400,len:5828
entry 0x400806a8
Ada
```


### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

The 0.14 still crash when unchecking all the boxes from  Sync Interface.
Also, can't compile the 0.15
`> node tools/cdata.js 

node:internal/modules/cjs/loader:1227
  throw err;
  ^

Error: Cannot find module 'inliner'
Require stack:
- C:\Users\DesktopTV\Desktop\WLED-0_15\tools\cdata.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1224:15)
    at Module._load (node:internal/modules/cjs/loader:1050:27)
    at Module.require (node:internal/modules/cjs/loader:1310:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (C:\Users\DesktopTV\Desktop\WLED-0_15\tools\cdata.js:20:17)
    at Module._compile (node:internal/modules/cjs/loader:1480:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1564:10)
    at Module.load (node:internal/modules/cjs/loader:1287:32)
    at Module._load (node:internal/modules/cjs/loader:1103:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:168:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'C:\\Users\\DesktopTV\\Desktop\\WLED-0_15\\tools\\cdata.js' ]
}

Node.js v22.1.0
*** Error 1`

--- a/wled00/FX.cpp
+++ b/wled00/FX.cpp
@@ -3547,7 +3547,7 @@ uint16_t mode_exploding_fireworks(void)
   if (segs <= (strip.getMaxSegments() /4)) maxData *= 2; //ESP8266: 1024 if <= 4 segs ESP32: 2560 if <= 8 segs
   int maxSparks = maxData / sizeof(spark); //ESP8266: max. 21/42/85 sparks/seg, ESP32: max. 53/106/213 sparks/seg
 
-  unsigned numSparks = min(2 + ((rows*cols) >> 1), maxSparks);
+  unsigned numSparks = min(5 + ((rows*cols) >> 1), maxSparks);
   unsigned dataSize = sizeof(spark) * numSparks;
   if (!SEGENV.allocateData(dataSize + sizeof(float))) return mode_static(); //allocation failed
   float *dying_gravity = reinterpret_cast<float*>(SEGENV.data + dataSize);
@@ -3602,7 +3602,8 @@ uint16_t mode_exploding_fireworks(void)
      * Size is proportional to the height.
      */
     unsigned nSparks = flare->pos + random8(4);
-    nSparks = constrain(nSparks, 4, numSparks);
+    nSparks = std::max(nSparks, 4U);  // This is not a standard constrain; numSparks is not guaranteed to be at least 4
+    nSparks = std::min(nSparks, numSparks);
 
     // initialize sparks
     if (SEGENV.aux0 == 2) {

